src/lib/operators/aggregate_hash.cpp

```diff
@@ -624,10 +628,7 @@ std::shared_ptr<const Table> AggregateHash::_on_execute() {
     }
     _write_groupby_output(pos_list);
   }
-
-  // _aggregate and _write_groupby_output have their own, internal timer. Start measuring once they are done.
-  auto& step_performance_data = static_cast<OperatorPerformanceData<OperatorSteps>&>(*performance_data);
-  Timer timer;
+  step_performance_data.set_step_runtime(OperatorSteps::GroupByColumnsWriting, timer.lap());
```

Comment:
Have a look at what your timer measures. If `_has_aggregate_functions` is true, no code is executed between the creation of the timer and `timer.lap()`.

`_write_groupby_output` is also called from `_write_aggregate_output`.

Reply:
Changed that. Not sure how happy I am with this solution.

Reply:
Have a look at #2270, where I ran into a similar problem. We should decide for one solution.

Reply:
Doesn't look the same to me as you are adding step times within another function, aren't you? I don't want to pass runtimes between functions.

Reply:
Just reviewed and saw that you are using members. I'll think about it...